common/app/model/ShareLinks.scala

```diff
@@ -93,7 +93,7 @@ object ShareLinks {
     lazy val facebookParams = List(
       Some("app_id" -> facebookAppId),
       Some("href" -> encodedHref),
-      quote.map(q => "quote" -> q),
+      quote.map(q => "quote" -> q.urlEncoded),
       mediaPath.map(path => "picture" -> path.urlEncoded)
```

Comment:
Basically, if you don't urlencode this part of it, then the you get an error like:

    java.net.URISyntaxException: Illegal character in query at index 194: https://www.facebook.com/dialog/share?app_id=157221981065719&href=https%3A%2F%2Fdiscussion.code.dev-theguardian.com%2Fcomment-permalink%2F64283254&quote=niclong+commented:+%22breaking+things+%7B"

I've managed to replicate the error locally. 

So the bug happens:

- when social sharing links are ON for comments
- and the discussion contains at least one comment in the comments being requested that contains an invalid character

where 'invalid character' means anything which is not allowed in URLs and not encoded.